scripts/update_toyota_software_version

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+from hero_bringup.toyota_software_utils import (
+    get_compatible_toyota_software_version_file,
+    get_installed_toyota_pkgs,
+    pkgs_to_version_dict,
+    read_dict_yaml,
+)
+import os
+import sys
+import yaml
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Set the current installed Toyota software as compatible")
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "-r",
+        "--reset",
+        action="store_true",
+        help="Current installed versions overwrite the stored compatible versions",
+    )
+    group.add_argument(
+        "-a", "--add", action="store_true", help="Add the current versions to the stored compatible versions"
+    )
+
+    args = parser.parse_args()
+
+    # Get all Toyota packages, so containing either hsr or tmc in the name
+    toyota_pkgs = get_installed_toyota_pkgs()
+
+    # Get the installed version of all Toyota packages
+    installed_pkgs = pkgs_to_version_dict(toyota_pkgs, print)
+
+    # Determine compatible software version file
+    compatible_pkgs_file = get_compatible_toyota_software_version_file()
+
+    # Read current set of compatible versions in case it doesn't need to be reset and the file exists
+    if args.reset or not os.path.isfile(compatible_pkgs_file):
+        compatible_pkgs = {}
+        # Set so final print statement is correct
+        args.add = False
+        args.reset = True
+    elif args.add:
+        # Read file with compatible Toyota software versions
+        compatible_pkgs = read_dict_yaml(compatible_pkgs_file, print)
+
+    # Add the package if not registered yet; Add current version to list of compatible versions in case package is
+    # already registered
+    for pkg_name, pkg_version in installed_pkgs.items():
+        if pkg_name not in compatible_pkgs.keys():
+            compatible_pkgs.update({pkg_name: [pkg_version]})
```

Comment:
Why map to a list? I can imagine that it might be convenient initially (as long as the stuff is consistent) but if it breaks you need to remove stuff entirely. I'm not sure what would be convenient.

Reply:
Yes, in that case you need to use the `--reset` option. Which resets to only the currently installed packages.